src/App.jsx

```diff
@@ -14,45 +16,102 @@ export const goodsFromServer = [
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORTED_BY_ALPHABET = 'name';
+const SORTED_BY_LENGTH = 'length';
+
+const getPreparedGoods = (goods, { sortField, isReversed }) => {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORTED_BY_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SORTED_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const changedGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
+  const clearSort = () => {
+    setSortField('');
+    setIsReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(SORTED_BY_ALPHABET)}
+          type="button"
+          className={cn('button', {
+            'is-info': sortField === SORTED_BY_ALPHABET,
+            'is-light': sortField !== SORTED_BY_ALPHABET,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortField(SORTED_BY_LENGTH)}
+          type="button"
+          className={cn('button', {
+            'is-success': sortField === SORTED_BY_LENGTH,
+            'is-light': sortField !== SORTED_BY_LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setIsReversed(reversed => !reversed)}
+          type="button"
+          className={cn('button', {
+            'is-warning': isReversed,
+            'is-light': !isReversed,
+          })}
+        >
+          Reverse
+        </button>
+
+        {sortField || isReversed ? (
+          <button
+            onClick={clearSort}
+            type="button"
+            className="button is-danger is-light"
+          >
+            Reset
+          </button>
+        ) : null}
```

Comment:
we can simplify this logic using the && instead of the ternary operator

Reply:
I fixed it, but here is such an error and I don't understand how to fix it? {!sortField && !isReversed && (
          <button
            onClick={clearSort}
            type="button"
            className="button is-danger is-light"
          >
            Reset
          </button>
        )}

Reply:
{sortField || isReversed && (
          <button
            onClick={clearSort}
            type="button"
            className="button is-danger is-light"
          >
            Reset
          </button>
        )}